File: prototype-parts-marking-development/src/client-app/src/models/api/items/part/prototype.ts
import { IUserData } from '../../users/userData';

export interface IPrototype
{
  id: number;
  prototypeSetId: number;
  type: string;
  partTypeCode: string;
  partTypeTitle: string;
  index: number;
  materialNumber: string;
  revisionCode: string;
  owner: IUserData;
  comment: string;
  createdAt: Date;
  createdBy: IUserData;
  modifiedAt: Date;
  modifiedBy: IUserData;
  deletedAt: Date;
  deletedBy: IUserData;
}

File: prototype-parts-marking-development/src/WebApi/Data/Prototype.cs
namespace WebApi.Data
{
    using System;
    using System.Collections.Generic;

    public class Prototype : IAuditableEntity
    {
        public int Id { get; set; }

        public int PrototypeSetId { get; set; }

        public PrototypeSet PrototypeSet { get; set; }

        public string PartTypeCode { get; set; }

        public string PartTypeTitle { get; set; }

        public PrototypeType Type { get; set; }

        public int Index { get; set; }

        public string Comment { get; set; }

        public string MaterialNumber { get; set; }

        public string RevisionCode { get; set; }

        public int OwnerId { get; set; }

        public User Owner { get; set; }

        public List<PrototypeVariant> PrototypeVariants { get; set; }

        public DateTimeOffset? CreatedAt { get; set; }

        public int CreatedById { get; set; }

        public User CreatedBy { get; set; }

        public DateTimeOffset? ModifiedAt { get; set; }

        public int ModifiedById { get; set; }

        public User ModifiedBy { get; set; }

        public DateTimeOffset? DeletedAt { get; set; }

        public int? DeletedById { get; set; }

        public User DeletedBy { get; set; }
    }
}
